### PR TITLE
Check the right matches in lambdas

### PR DESCRIPTION
**Pull request by davidwengier**
Fixes https://github.com/dotnet/roslyn/issues/61415

### CODE REVIEW

File: src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs [hunk 1]
```diff
@@ -1162,7 +1165,7 @@ public LambdaInfo WithMatch(Match<SyntaxNode> match, SyntaxNode newLambdaBody)
 
                     // exception handling around the statement:
                     CalculateExceptionRegionsAroundActiveStatement(
-                        bodyMatch,
+                        match,
```
Comment (davidwengier) [inline src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs:1168]: Could also have removed line 1150 where I set `match = bodyMatch` and instead made this `hasMatching ? match! : bodyMatch` or even `match ?? bodyMatch`. This way seemed a little more future proof.

**Comment by davidwengier** [comment]
@tmat

File: src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs [hunk 2]
```diff
@@ -9930,6 +9930,56 @@ async Task<int> f()
                 Diagnostic(RudeEditKind.UpdatingStateMachineMethodAroundActiveStatement, "f"));
         }
 
+        [Fact]
+        [WorkItem(61415, "https://github.com/dotnet/roslyn/issues/61415")]
+        public void LocalFunction_EditAroundTry_WithActiveStatement()
+        {
+            var src1 =
+                """
```
Comment (tmat) [inline src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9938]: >""" [](http://example.com/codeflow?start=16&length=3)

nit: do raw strings actually make this better with all the indentation? Maybe I just got used to the previous style.

File: src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs [hunk 3]
```diff
@@ -9930,6 +9930,56 @@ async Task<int> f()
                 Diagnostic(RudeEditKind.UpdatingStateMachineMethodAroundActiveStatement, "f"));
         }
 
+        [Fact]
+        [WorkItem(61415, "https://github.com/dotnet/roslyn/issues/61415")]
+        public void LocalFunction_EditAroundTry_WithActiveStatement()
```
Comment (tmat) [inline src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9935]: >LocalFunction_EditAroundTry_WithActiveStatement [](http://example.com/codeflow?start=20&length=47)

This issue also repros for lambdas, not just local functions, right? Would you mind adding another test to cover that?

File: src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs [hunk 4]
```diff
@@ -1162,7 +1165,7 @@ public LambdaInfo WithMatch(Match<SyntaxNode> match, SyntaxNode newLambdaBody)
 
                     // exception handling around the statement:
                     CalculateExceptionRegionsAroundActiveStatement(
```
Comment (tmat) [inline src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs:1167]: >CalculateExceptionRegionsAroundActiveStatement [](http://example.com/codeflow?start=20&length=46)

If we don't have `match` then that means the lambda containing the AS was removed (or we couldn't find it). If that is the case the exception region analysis does not make much sense, so I think we should skip it.

File: src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs (hunk 2 above)
Comment (davidwengier) [reply src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9938]: In this case, with no quotes in the string, the only thing the raw string is providing is that the test input doesn't have to be based on column 0, so flows better in the IDE. The extra indentation is not present in the actual test input of course.

Comment (tmat) [reply src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9988]: Of course, though "flows better" is subjective here. I feel like I need to subtract the whitespace in my head when reading the embedded source code.

Comment (davidwengier) [reply src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9988]: Yeah, thats a good point. I'm very used to the vertical line in VS that takes away that pain.

Comment (tmat) [reply src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs:9988]: Yup, it's better in VS but sadly github/CodeFlow do not have it :(

**Comment by davidwengier** [comment]
ping @tmat

**Review by tmat — APPROVED** [review]
:shipit:

**Pull request merged**